run-in-terminal@alextpedro/files/run-in-terminal@alextpedro/run_term.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+#x-terminal-emulator
+DIR="$( cd "$( dirname "$1" )" && pwd )"
+gnome-terminal -- bash -c "cd $DIR; $1; exec bash"
+
```

Comment:
There's a few pieces here:

* The first is that the terminal is hard-coded but it might not be installed or not the user's default terminal.
* For gnome-terminal and a few other terminals that have adopted it, you can pass in a `--working-directory` parameter (but using the GLib function to spawn the command is likely a safer structure with the additional complication notwithstanding).
* The code here doesn't match the action file. You either want to set Selection to `s` or to execute multiple scripts here if there are any. You might also want to add a check script to make sure that the script is executable (or update your invocation of it).
* The workaround to keep the shell open could maybe use the default shell of the user in case it isn't bash but this is a minor point compared to the previous ones.

Reply:
I tried to make it more generic by using x-terminal-emulator but all I managed to get going with that was to open a terminal window in the home directory, it seems to ignore everything in front. I couldn't pass the script to it or anything. I've also seen xdg-terminal referred to around the web but it doesn't seem to be a thing on Debian-based distros? I also don't know how to get the default shell.

Reply:
Right, each distro has their own convention. There is a gsetting you can retrieve:
```text
gsettings get org.cinnamon.desktop.default-applications.terminal exec
```

The only assumption this makes is that Cinnamon is installed which technically isn't always guaranteed with Nemo (and therefore someone could install Actions manually and encounter problems) but that is a separate issue. To be frank, that has been the challenge with a few of the attempts at Actions like this. The challenge after this is that there isn't a universal way to specify the working directory cleanly (even spawning with GLib has an issue with default terminals from Flatpak).

Reply:
I've made some changes to attempt to use the user's default terminal and shell, but I am unsure if what I did works in that way. 

I'm also not entirely sure what is wrong with the way I am fetching the directory. 

As for whether the script is executable, from my tests it will simply open the terminal with the message "permission denied". I also have no idea how to check that.

Reply:
> As for whether the script is executable, from my tests it will simply open the terminal with the message "permission denied". I also have no idea how to check that.

This would be a helper script. Take a look at a clone-git-repo@anaximeno which was recently merged for an example.

I don't mean to make this more complicated for you so I apologize for that. I think the reason we haven't seen a lot of Actions like these are because they tend to be very difficult to implement.

Reply:
Finally had time to add the helper script! 

It's fine, but this is definitely getting over the scope of what I was trying to do! I think it's fine to have actions that only work in specific circumstances, since these are not supposed to be for every user.